### PR TITLE
[2830] Add GOG Galaxy session support

### LINKED ISSUE #2830: [CHARACTER CREATION] Provide an alternate way to assign characters for GOG players
# For New Contributors

[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

Not provided

## Description

GOG players can join a dedicated server, but because GOG does not have Steam IDs, two players are assigned the same character (character 0).

## Intended Design

Not provided

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

Not provided

## Related Issues

Not provided

## Requirements

Not provided

## Additional information

Not provided

## Definition of Done

- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

Not provided

## Discord source

https://discord.com/channels/709516043332354119/1535499913369231370/1535499913369231370

### PR DESCRIPTION
**Pull request by ShoT-UPfps**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behavior?

GOG Galaxy players cannot host, browse, or join co-op games through Galaxy. GOG player IDs can be supplied by the client and can collide with IDs from other platforms.

## What is the new behavior?

Resolves #2830

GOG Galaxy players now see a Galaxy lobby browser and can host or join Galaxy sessions, while Steam players continue to use Steam. The server gets each GOG player's identity from their Galaxy connection and keeps GOG IDs separate from Steam IDs.

## Bot Changelog Entry

- Added GOG Galaxy lobby browsing, hosting, and joining with connection-bound player identities.

### CODE REVIEW

File: source/Coop.GOG/GalaxySdk.cs [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+    public void CreateLobby(
+        GalaxyLobbyVisibility visibility,
+        int maxMembers,
+        Action<ulong, bool> onCompleted)
+    {
+        if (onCompleted == null) throw new ArgumentNullException(nameof(onCompleted));
+
+        var listener = new LobbyCreatedListener(onCompleted);
+        matchmaking.CreateLobby(
+            gameServer ? LobbyType.LOBBY_TYPE_PUBLIC : ToLobbyType(visibility),
```
Comment (gamingmenacers-svg) [inline source/Coop.GOG/GalaxySdk.cs:102]: this replaces `FriendsOnly` with `LOBBY_TYPE_PUBLIC` for every dedicated server. `RequestLobbyList` only filters on the tunnel provider and `GalaxyLobbyBrowser.TryBuildListing` never enforces `coop_visibility`, so a friends-only dedicated server is publicly enumerable and joinable. preserve the requested lobby type here, or enforce the friends-only policy before returning listings/allowing joins, and add a test through the real game-server adapter rather than only the fake sdk boundary.

**Review by gamingmenacers-svg — CHANGES_REQUESTED** [review]
reviewed this against current `development`, including the provider abstraction, galaxy lobby lifecycle, p2p fragmentation/tunnels, identity bridge, dedicated-server boot, packaging, and the new tests.

there is one release-blocking visibility hole in the real sdk adapter. the unit tests only assert the visibility passed into `IGalaxySdk`, so they do not exercise the game-server branch that discards it. a dedicated GOG server configured `FriendsOnly` creates a public lobby, and `GalaxyLobbyBrowser` accepts that listing without checking `coop_visibility`, making it discoverable to unrelated GOG users. `None` happens to be hidden by the separate listing-type value, but friends-only is not enforced anywhere after this point.

this branch is also 256 commits behind and currently conflicting, with build/test jobs skipped and CodeFactor failing, so the full provider matrix needs to run after the visibility fix and rebase.
